[docs][PLAT-7630] Troubleshooting section on Kubernetes debug hooks (#16575)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/troubleshoot/universe-issues.md

                "level": "info",
                "value": 2048
            },

> …

Reviewer: ```suggestion
...
```

Author: Reverted it. I thought these three dots are causing GitHub to highlight half of the page as red. That's not the case, don't know why it happens with some pages.

  The **Support Bundles** dialog allows you to either download the bundle or delete it if it is no longer needed. By default, bundles expire after ten days to free up space.

## Debug crashing YugabyteDB pods in Kubernetes

If the YugabyteDB pods of your universe are crashing, you can debug them with the help of following instructions.

### Collect core dumps in Kubernetes environments

> When dealing with Kubernetes-based installations of YugabyteDB Anywhere, you might need to retrieve core dump files in case of a crash within the Kubernetes pod. For more information, see [Specify ulimit and remember the location of core dumps](../../install-yugabyte-platform/prerequisites#specify-ulimit-and-remember-the-location-of-core-dumps).

Reviewer: links to wrong location

Author: Yes, I fixed it my other PR, but let me fix here as we will merge this one first.

- If the value of `core_pattern` contains a `|` pipe symbol (for example, `|/usr/share/apport/apport -p%p -s%s -c%c -d%d -P%P -u%u -g%g -- %E`), the core dump is being redirected to a specific collector on the underlying Kubernetes node, with the location depending on the exact collector. In this case, it is your responsibility to identify the location to which these files are written and retrieve them.

### Use debug hooks with YugabyteDB in Kubernetes

> You can add your own commands to pre and post debug hooks to troubleshoot crashing YB-Master or YB-TServer pods. These commands are run before the database process starts and after the database process terminates or crashes.

Reviewer: ```suggestion
You can add your own commands to pre- and post-debug hooks to troubleshoot crashing YB-Master or YB-TServer pods. These commands are run before the database process starts and after the database process terminates or crashes.
```

> For example, to modify the debug hooks of a YB-Master, you will run following commands:

Reviewer: ```suggestion
For example, to modify the debug hooks of a YB-Master, run following command:
```

```sh
kubectl edit configmap -n <namespace> ybuni1-asia-south1-a-lbrl-master-hooks
```

> This will open the configmap YAML in your editor.

Reviewer: ```suggestion
This opens the configmap YAML in your editor.
```

> To add multiple commands to pre hook of `yb-master-0`, you can modify the `yb-master-0-pre_debug_hook.sh` key as follows:

Reviewer: ```suggestion
To add multiple commands to the pre-debug hook of `yb-master-0`, you can modify the `yb-master-0-pre_debug_hook.sh` key as follows:
```

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: ybuni1-asia-south1-a-lbrl-master-hooks
data:
  yb-master-0-post_debug_hook.sh: 'echo ''hello-from-post'' '
  yb-master-0-pre_debug_hook.sh: |
    echo "Running the pre hook"
    du -sh /mnt/disk0/yb-data/
    sleep 5m
    # other commands here…
  yb-master-1-post_debug_hook.sh: 'echo ''hello-from-post'' '
  yb-master-1-pre_debug_hook.sh: 'echo ''hello-from-pre'' '
```

> Once you save the file, the updated commands will be executed on the next restart of `yb-master-0`.

Reviewer: ```suggestion
After you save the file, the updated commands will be executed on the next restart of `yb-master-0`.
```

You can run the following command to check the output of your debug hook:

```sh
kubectl logs -n <namespace> ybuni1-asia-south1-a-lbrl-yb-master-0 -c yb-master
```

Expect an output similar to the following:

```output

> …

Reviewer: ...
use 3 dots instead of an ellipsis character

Author: Okay I will. Just curios: what is the reason to not use ellipsis?

Reviewer: I thought it was causing this annoying red highlighting when reviewing, but it's still there with three dots so I'm not sure what the deal is. They both render the same on the actual site.
